docs/src/design/proposals/APIRedesign.md

```diff
@@ -0,0 +1,1050 @@
+  understood in industry. This can be beneficial in training
+  new staff, and lowers the documentation burden on the local
+  development team. If MCT-Web were to stay with its current
+  architecture, significant developer-oriented documentation
+  and training materials would need to be produced.
+* The maintainability of MCT-Web would be enhanced by using a
+  framework like Angular. The local team would enjoy the benefits of
+  maintenance performed by the sponsor, but would not incur any cost
+  for this. This would include future upgrades, testing, and bug fixes.
+* Replaces DOM-manipulation with a declarative data-binding syntax
+  which automatically updates views when the model data changes. This
+  pattern has the potential to save the development team from
+  time-consuming and difficult-to-debug DOM manipulation.
+* Provides data binding to backend models.
+* Provides patterns for form validation.
+* Establishes documented patterns for add-on modules and services.
+* Supports unit tests and system tests (tests which simulate user
+  interactions in the browser)
+* Angular software releases can be expected to be tested, which would
+  allow MCT-Web developers to focus on MCT-specific features, instead
+  of the maintenance of custom infrastructure.
+
+### Actual Experience with Angular
+
+Most of the expected benefits of Angular have been invalidated
+by experience:
+
+* Feedback from new developers is that Angular was a hindrance to
+  training, not a benefit. ("One more thing to learn.") Significant
+  documentation remains necessary for Open MCT Web.
+* Expected enhancements to maintainability will be effectively
+  invalidated by an expected Angular end-of-life.
+* Data binding and automatic view updates do save development effort,
+  but also carry a performance penalty. This can be solved, but requires
+  resorting to exactly the sort of DOM manipulations we want to avoid.
+  In some cases this can require more total development (writing a
+  poorly-performing Angular version, then "optimizing" by rewriting a
+  non-Angular version.)
+* Expected reduction of test scope will also be invalidated by an
+  expected end-of-life.
+
+Other problems:
+
+* Hinders integrating non-Angular components. (Need to wrap with
+  Angular API, e.g. as directives, which may be non-trivial.)
+* Interferes with debugging by swallowing or obscuring exceptions.
+
+# Feedback Review
+
+## Problem Summary
+
+The following attributes of the current API are undesirable:
+
+- [ ] It is difficult to tell "where things are" in the code base.
+- [ ] It is difficult to see how objects are passed around at run-time.
+- [ ] It is difficult to trace flow of control generally.
+- [ ] Multiple interfaces for related concepts (e.g. telemetry) is confusing.
+- [ ] API documentation is missing or not well-formatted for use.
+- [ ] High-level separation of concerns is not made clear.
+- [ ] Interface depth of telemetry API is excessive (esp. `TelemetrySeries`)
```

Comment:
I would argue that it's not just the depth of the API, depth aside, the overall path from telemetry provider to view is not obvious.